frontend/src/pages/auth/signin/SignIn.tsx

```diff
@@ -47,9 +60,12 @@ const SignIn: React.FC = () => {
         const decodedToken: TokenData = jwtDecode(token);
         const user = await getUserService(decodedToken.userId, decodedToken.userType);
         const loggedUser = createLoggedUser(token, user);
-        if (loggedUser) {
-            dispatch(signIn(loggedUser));
-        }
+        if (!loggedUser) return;
+        const wallet = await getWalletByUserService(loggedUser.id, loggedUser.userType);
+        const loggedWallet = createIWallet(wallet);
```

Comment:
ah, I see, you are not creating here wallet. You are mapping one model to another. Your rule number 1 in use.
Just remove it and use `IWallet`, your interface will not allow to use other fields anyway

Reply:
After you do it, think again, why do you need to get wallet after login? 
Let's say, user logged in, then they go to the main page and see the wallet. It's ok for now. During the day, they make some payments. Then go to main page again and see that balance is same as before. Looks like a perfect wallet which never goes down.

Reply:
1) if i don't store in redux, i have to every single time call `getWalletService` to get data about wallet when i navigate to the Home page
2) when i provided payment or top up, i will get fresh data about my wallet to update data in redux

Reply:
I see, exactly in such flow it works ok.  But if user gets money from somewhere else or make a payment from another device your current state does not know anything about it. Fundamentally, it's a wrong approach.  
You have to get balance in 2 cases:
1) All the time when user reloads a page
2) Either each time when user goes to balance page ( I prefer it, since it's same as one above)
or you have to add additional button to manually reload balance but still support first point